hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -335,7 +337,18 @@ public interface HoodieTimeline extends Serializable {
   /**
    * @return Get the stream of completed instants
    */
-  Stream<HoodieInstant> getInstants();
+  default Stream<HoodieInstant> getInstantsAsStream() {
+    return getInstants().stream();
+  }
+
+  /**
+   * @return Get tht list of instants
+   */
+  List<HoodieInstant> getInstants();
+
+  default List<HoodieInstant> getAndCopyInstants() {
+    return new ArrayList<>(getInstants());
+  }
```

Comment:
ditto

also we should use `new ArrayList<>` for `getInstants()` and never expose `instants` to users. so we should just have `getInstants`, and make javadoc clear about the behavior and usage

Reply:
agreed.